components/CampaignCard.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from 'react';
+import { faDonate } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const CampaignCard = ({ name, text, logoUrl, donate }) => {
+    const [state, setstate] = useState(false);
+    const [details, setDetails] = useState('');
+
+    useEffect(() => {
+        setDetails(`${text.substr(0, 300)}...`);
+    }, [text]);
+
+    const textLen = () => {
+        setstate(!state);
+        if (state) {
+            setDetails(text);
+        } else {
+            setDetails(`${text.substr(0, 300)}...`);
+        }
+    };
+    return (
+        <div className="w-full bg-white rounded-lg shadow dark:bg-gray-1200 dark:text-gray-300 p-4 my-3">
+            <div className="flex justify-between flex-wrap mb-4">
+                <div className="font-bold text-3xl dark:text-white">{name}</div>
+                <div>
+                    <img src={logoUrl} alt="logo" className="h-8" />
+                </div>
+            </div>
+            <div dangerouslySetInnerHTML={{ __html: details }}></div>
```

Comment:
🤔🤔 `dangerouslySetInnerHTML` would not be better choice maybe. 
Make a quick migration with https://github.com/tasti/react-linkify/